examples/userapi/04_boundary_conditions.ipynb

```diff
@@ -0,0 +1,455 @@
+      "text/plain": [
+       "<Figure size 432x288 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "src.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For our PMLs, we will need some damping parameter. In this case, we will use a quadratic taper over the absorbing regions on the left and right sides of the domain."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Damping parameterisation\n",
+    "damp_l = (1-0.1*x)**2  # Left side\n",
+    "damp_r = (1-0.1*(grid.shape[0]-x))**2  # Right side"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now for our main domain equations:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from devito import Eq, grad, div\n",
+    "\n",
+    "eq_v = Eq(v.forward, v - dt*grad(u)/density,\n",
+    "          subdomain=grid.subdomains['main'])\n",
+    "\n",
+    "eq_u = Eq(u.forward, u - dt*velocity**2*density*div(v.forward),\n",
+    "           subdomain=grid.subdomains['main'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "And the equations for our damped region:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Left side\n",
+    "eq_v_damp_left = Eq(v.forward,\n",
+    "                    (1-damp_l)*v - dt*grad(u)/density,\n",
+    "                    subdomain=grid.subdomains['left'])\n",
+    "\n",
+    "eq_u_damp_left = Eq(u.forward,\n",
+    "                    (1-damp_l)*u - dt*velocity**2*density*div(v.forward),\n",
+    "                     subdomain=grid.subdomains['left'])\n",
+    "\n",
+    "# Right side\n",
+    "eq_v_damp_right = Eq(v.forward,\n",
+    "                     (1-damp_r)*v - dt*grad(u)/density,\n",
+    "                     subdomain=grid.subdomains['right'])\n",
+    "\n",
+    "eq_u_damp_right = Eq(u.forward,\n",
+    "                    (1-damp_r)*u - dt*velocity**2*density*div(v.forward),\n",
+    "                     subdomain=grid.subdomains['right'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Add our free surface boundary conditions (currently only Drichlet...):"
```

Comment:
That is not a free surface, this is a zero Dirichlet boundary, so would not call it like that. Free surface requires to mirror `vz`

Reply:
Yeah, this is on my to do list. I figured I'd get it in a working state for review, but I haven't got around to finishing this bit yet